src/Tools/dotnet-openapi/src/Commands/BaseCommand.cs

```diff
@@ -0,0 +1,308 @@
+            {
+                var args = new string[] {
+                    "add",
+                    "package",
+                    packageId,
+                    "--version",
+                    version,
+                    "--no-restore"
+                };
+
+                var muxer = DotNetMuxer.MuxerPathOrDefault();
+                if(string.IsNullOrEmpty(muxer))
+                {
+                    throw new ArgumentException($"dotnet was not found on the path.");
+                }
+
+                var startInfo = new ProcessStartInfo
+                {
+                    FileName = muxer,
+                    Arguments = string.Join(" ", args),
+                    WorkingDirectory = projectFile.Directory.FullName,
+                    RedirectStandardError = true,
+                    RedirectStandardOutput = true,
+                };
+
+                var process = Process.Start(startInfo);
+
+                if (!process.WaitForExit(20*1000))
+                {
+                    throw new ArgumentException($"Adding package `{packageId}` to `{projectFile.Directory}` took too long.");
+                }
+
+                if (process.ExitCode != 0)
+                {
+                    Error.Write(process.StandardError.ReadToEnd());
+                    Error.Write(process.StandardOutput.ReadToEnd());
+                    throw new ArgumentException($"Could not add package `{packageId}` to `{projectFile.Directory}`");
+                }
+            }
+        }
+
+        internal string GetFullPath(string path)
+        {
+            return Path.IsPathFullyQualified(path)
+                ? Path.GetFullPath(path)
+                : Path.GetFullPath(path, WorkingDirectory);
+        }
+
+        private static IEnumerable<Tuple<string, string>> GetServicePackages(CodeGenerator type)
+        {
+            var name = Enum.GetName(typeof(CodeGenerator), type);
+            var attributes = typeof(Program).Assembly.GetCustomAttributes<AssemblyMetadataAttribute>();
+            var attribute = attributes.Single(a => string.Equals(a.Key, name, StringComparison.OrdinalIgnoreCase));
+
+            var packages = attribute.Value.Split(';', StringSplitOptions.RemoveEmptyEntries);
+            var result = new List<Tuple<string, string>>();
+            foreach (var package in packages)
+            {
+                var tmp = package.Split(':', StringSplitOptions.RemoveEmptyEntries);
+                Debug.Assert(tmp.Length == 2);
+                result.Add(new Tuple<string, string>(tmp[0], tmp[1]));
+            }
+
+            return result;
+        }
+
+        private async Task WriteToFile(Stream content, string destinationPath, bool overwrite)
+        {
+            var destinationExists = File.Exists(destinationPath);
+            if (destinationExists && !overwrite)
+            {
+                await Out.WriteLineAsync($"Not overwriting existing file '{destinationPath}'.");
```

Comment:
Why the WriteLineAsync now? You've used WriteLine in a bunch of other places.

Reply:
I decided not to make methods async just for these WriteLines, so in those methods that didn't otherwise go async yes, it's mixed. I did a quick scrub for WriteLines violating this idea and found a couple.